### PR TITLE
Rdm 7419: Introduce Cucumber Runner Classes

### PR DESCRIPTION
**Pull request by kiran-yenigala-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-7419

### Change description ###

Introduce Cucumber Runner Class

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreBeftaRunner.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import uk.gov.hmcts.befta.BeftaMain;
+import uk.gov.hmcts.befta.BeftaTestDataLoader;
+import uk.gov.hmcts.befta.DefaultTestAutomationAdapter;
+
+
+@RunWith(Cucumber.class)
+@CucumberOptions(plugin = "json:target/cucumber.json",
+    glue = "uk.gov.hmcts.befta.player",
+    features = {"classpath:features"})
+public class DataStoreBeftaRunner extends BeftaMain {
```
Comment (MSancaktutar) [inline src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreBeftaRunner.java]: Could we please not extend BeftaMain, as this class wouldn't reasonably form an is-a relation with it?

File: src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreBeftaRunner.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    @BeforeClass
+    public static void init() {
+        setTaAdapter(new DefaultTestAutomationAdapter());
+        BeftaTestDataLoader.main(null);
```
Comment (MSancaktutar) [inline src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreBeftaRunner.java]: Apart from the data loading, there is also another tear-up logic to inject syntactic flexibilities into DSL elements. That is inside BeftaMain. We should probably combine any such initial tear up logic into a dedicated public static method inside befta main and re-use that from anaywhere needed. The adapted object can be a parameter to that one. This way, we can standardise and re-use the logic to max extent.

Comment (kiran-yenigala-hmcts) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreBeftaRunner.java]: so we need to change ccd-api-test-automation-fw for this change then

File: src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreBeftaRunner.java (hunk 1 above)
Comment (kiran-yenigala-hmcts) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreBeftaRunner.java]: Agree.. will remove that

File: src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreBeftaRunner.java (hunk 2 above)
Comment (MSancaktutar) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreBeftaRunner.java]: yep..

Comment (kiran-yenigala-hmcts) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreBeftaRunner.java]: Changed in ccd-api-test-automation-fw  and calling the necessary methods

**Pull request merged**